compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -56,6 +56,20 @@ impl<'tcx> Context for Tables<'tcx> {
         trait_def.stable(self)
     }
 
+    fn all_trait_impls(&mut self) -> stable_mir::ImplTraitDecls {
+        self.tcx
+            .trait_impls_in_crate(LOCAL_CRATE)
+            .iter()
+            .map(|impl_def_id| self.impl_def(*impl_def_id))
+            .collect()
+    }
+
+    fn trait_impl(&mut self, impl_def: &stable_mir::ty::ImplDef) -> stable_mir::ty::TraitRef {
+        let def_id = self.impl_trait_def_id(impl_def);
+        let impl_trait_def = self.tcx.impl_trait_ref(def_id).unwrap().skip_binder();
```

Comment:
@oli-obk given that this is an `EarlyBinder` I think we can just `skip_binder` here to simplify things.

Reply:
I believe what we want is `instantiate_identity()`, which signals that we are not ignoring arguments, but instead expect to be using it in the parameter environment of the parent.

That said, we should probably start preserving early binders